### DISCUSSION TITLE
Give field a default value if derived from an AbstractClass

### DISCUSSION THREAD

**Discussion by thomas-pedot** [pydantic/pydantic#3010] [Question] [answered]
Hello,
I write a class which derived from a abstracted class.

I want to set a field to a default value if not provided at instanciation. 

```
SOME_DEFAULT = 'mydefault'
class ModelAbstract(BaseModel, ABC):
   my_field: str

class Model(ModelAbstract):
    @validator('my_field')
     def val_my_field(cls, v, values, **kwargs):
        if v is None or v == '':
            v = SOME_DEFAULT 
        return v
```

Doing this is not validating `my_field`.
I have my expected behavior adding this in __init__
```
    def __init__(self, **data) -> None:
        super().__init__(**data)
        if self.my_field is None or self.my_field == '':
            self.my_field = SOME_DEFAULT 
```

But it would be better if this was managed by pydantic since it duplicates the code.

Any idea ?

This might be realted to this but I have no clue:

https://github.com/samuelcolvin/pydantic/issues/1223

**Comment by lesnik512** [comment, marked as answer]
Hello.
But why abstract class?

Seems like it can be done just like this:

```python
from pydantic import BaseModel

class Model(BaseModel):
    my_field: str = 'default'

assert Model().my_field == 'default'
assert Model(my_field='not default').my_field != 'default'
```

**Reply by thomas-pedot** [reply]
Hello,
I need an abstract class because there will be many models which derives from it.

**Reply by lesnik512** [reply]
But you can derive from any Pydantic model. It doesn't required to be abstract
